### ISSUE TITLE
Question regarding DockerHub tags and newer images

### ISSUE THREAD

**Issue by mwhite-ibm** [linuxkit/linuxkit#3708]
Hello, I have a question regarding the tags in DockerHub for the various LinuxKit container images.

From time to time, my colleagues and I will pick up the most recent version of `linuxkit/runc` and other images. Navigating to the [DockerHub tags page for `linuxkit/runc`](https://hub.docker.com/r/linuxkit/runc/tags?page=1&ordering=last_updated), I see an older `latest` image, etc., though scrolling to the bottom there are several recent tags that are **not rendered as links**.

My colleague and I noticed that `docker trust inspect --pretty linuxkit/runc` contains many (maybe all?) of the tags that **are** rendered as links, but none of the ones that aren't. This seems to imply that these are unsigned images, perhaps built and pushed for testing purposes. Even further, we get the following error when we attempt to build with one of the "unsigned" images:
```bash
FATA[0003] Failed to build init tarball from docker.io/linuxkit/runc:bf1e0c61fb4678d6428d0aabbd80db5ea24e4d4d: Could not pull image docker.io/linuxkit/runc:bf1e0c61fb4678d6428d0aabbd80db5ea24e4d4d: Trusted pull for docker.io/linuxkit/runc:bf1e0c61fb4678d6428d0aabbd80db5ea24e4d4d failed: No valid trust data for bf1e0c61fb4678d6428d0aabbd80db5ea24e4d4d 
```

Removing the following from our LinuxKit build YAML eliminates the error, but we'd refer to not do so unless necessary.
```yaml
trust:
  org:
    - linuxkit
```

Could you please explain why the more recent tags aren't links, ie. are these indeed untrusted images? Are these legitimate images for use, or were they built and pushed for testing purposes?

It's probably also worth noting that we are pinning LinuxKit to version 0.8 in our builds (as opposed to using `master`), as we have a fairly heavy dependency on Docker.

Thanks!

-Matt

**Comment by mwhite-ibm** [comment]
Along similar lines, I noticed [for `linuxkit/getty`](https://hub.docker.com/r/linuxkit/getty/tags?page=1&ordering=last_updated) that some "trusted" images seem to lack OS/Arch metadata, as well as size information.

**Comment by rn** [comment]
we are in a transition phase where we are/have disabled docker content trust. Some older images on hub are still signed and some older versions of the `linuxkit` binary still enforce it if specified in the YAML.

Can you try a more recent version of the `linuxkit` utility?

We would like to do a release soon but we are currently stuck on some changes to runc/containerd which seem to break certain images.

**Comment by mwhite-ibm** [comment]
@rn 

Thanks for the speedy response!

Ahhhh, I can see upon further inspection that the `disable-content-trust` option has been removed on `master`.

Regarding trying a more recent version, I tested LinuxKit from `master` with the `docker` option specified and confirmed that I was able to build without any warnings.

So, in short the images are intentionally unsigned and LinuxKit isn't currently verifying.

I think you cleared up my questions, thanks a ton! I'll go ahead and close this out, will open another ticket if further assistance is needed.
